### PR TITLE
Add Jenkinsfile

### PR DESCRIPTION
**Pull request by mrjgreen**
Added jenkins file, now using the docker agent and s3 pipeline plugin

### CODE REVIEW

**Comment by mrjgreen** [comment]
will squash on merge

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        sh 'git config --global core.logallrefupdates false'
+        
+        sh 'rm -rf node_modules/*'
+      }
+    }
+    
+    stage('Clone Dependencies') {
+      steps {
+        sh 'rm -rf public/brand modules'
+        checkout([$class: 'GitSCM', branches: [[name: '*/master']], extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: 'public/brand'], [$class: 'CloneOption', shallow: true]], userRemoteConfigs: [[credentialsId: 'GIT_DEPLOY_KEY', url: 'git@github.com:sharelatex/brand-sharelatex']]])
```
Comment (jpallen) [inline Jenkinsfile]: Are all of these `[]`s necessary? It looks like a lot can be optional: https://jenkins.io/doc/book/pipeline/jenkinsfile/#optional-step-arguments

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        checkout([$class: 'GitSCM', branches: [[name: '*/master']], extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: 'modules/learn-wiki'], [$class: 'CloneOption', shallow: true]], userRemoteConfigs: [[credentialsId: 'GIT_DEPLOY_KEY', url: 'git@bitbucket.org:sharelatex/learn-wiki-web-module.git']]])
+        checkout([$class: 'GitSCM', branches: [[name: '*/master']], extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: 'modules/templates'], [$class: 'CloneOption', shallow: true]], userRemoteConfigs: [[credentialsId: 'GIT_DEPLOY_KEY', url: 'git@github.com:sharelatex/templates-webmodule.git']]])
+        checkout([$class: 'GitSCM', branches: [[name: '*/master']], extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: 'modules/track-changes'], [$class: 'CloneOption', shallow: true]], userRemoteConfigs: [[credentialsId: 'GIT_DEPLOY_KEY', url: 'git@github.com:sharelatex/track-changes-web-module.git']]])
+      }
+    }
+    
+    stage('Install') {
+      steps {
+        sh 'mv app/views/external/robots.txt public/robots.txt'
+        sh 'mv app/views/external/googlebdb0f8f7f4a17241.html public/googlebdb0f8f7f4a17241.html'
```
Comment (jpallen) [inline Jenkinsfile]: This is putting stuff we only use internally into public code. Not that big a deal, but could consider running an 'install' script here, that is provided by the `app/views/external` module?

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        checkout([$class: 'GitSCM', branches: [[name: '*/master']], extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: 'modules/learn-wiki'], [$class: 'CloneOption', shallow: true]], userRemoteConfigs: [[credentialsId: 'GIT_DEPLOY_KEY', url: 'git@bitbucket.org:sharelatex/learn-wiki-web-module.git']]])
+        checkout([$class: 'GitSCM', branches: [[name: '*/master']], extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: 'modules/templates'], [$class: 'CloneOption', shallow: true]], userRemoteConfigs: [[credentialsId: 'GIT_DEPLOY_KEY', url: 'git@github.com:sharelatex/templates-webmodule.git']]])
+        checkout([$class: 'GitSCM', branches: [[name: '*/master']], extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: 'modules/track-changes'], [$class: 'CloneOption', shallow: true]], userRemoteConfigs: [[credentialsId: 'GIT_DEPLOY_KEY', url: 'git@github.com:sharelatex/track-changes-web-module.git']]])
```
Comment (jpallen) [inline Jenkinsfile]: This 'leaks' quite a lot of information about how we configure SL in production, into the public repo. I can't really see a problem, it's just not that 'clean' in separating the two.

**Pull request merged**